packages/workspace/src/workspace/path_index.ts

```diff
@@ -108,6 +108,20 @@ export const updatePathIndex = (current: PathIndex, unmergedElements: Element[],
 }
 
 export const deserializedPathIndex = (data: string): PathIndex => new PathIndex(JSON.parse(data))
+export const deserializedPathsIndex = (dataEntries: string[]): PathIndex => new PathIndex(
+  _.flatten(dataEntries.map(data => JSON.parse(data)))
+)
 export const serializedPathIndex = (index: PathIndex): string => (
   safeJsonStringify(Array.from(index.entries()))
 )
+export const serializePathIndexByService = (index: PathIndex): Record<string, string> => {
+  const serviceToSerializedPathIndex: Record<string, [string, Path[]][]> = {}
+  Array.from(index.entries()).forEach(entry => {
+    const { adapter } = ElemID.fromFullName(entry[0])
+    if (!(adapter in serviceToSerializedPathIndex)) {
+      serviceToSerializedPathIndex[adapter] = []
+    }
+    serviceToSerializedPathIndex[adapter].push(entry)
+  })
+  return _.mapValues(serviceToSerializedPathIndex, value => safeJsonStringify(value))
```

Comment:
Style suggestion. Anyway please use `wu` instead of `Array.from`
```suggestion
Object.fromEntries(
  wu(index.entries())
    .map(entry => {
      const { adapter } = ElemID.fromFullName(entry[0])
      if (!(adapter in serviceToSerializedPathIndex)) {
        return undefined
      }
      return [adapter, safeJsonStringify(entry)]
    })
    .filter(isDefined)
)
```

Reply:
I'll use wu, but the code you suggest won't work, because I need to divide by keys, but I changed this to groupby, which will look better

Reply:
I simplified the code using groupBy, but that means I need to use Array.from. It seems fine to me since that's what the original code (see serializedPathIndex) uses